kv/kvserver/asim/tests: TestDataDriven failed

kv/kvserver/asim/tests.TestDataDriven [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_NightlyCockroachExtraStress/20287374?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_NightlyCockroachExtraStress/20287374?buildTab=artifacts#/) on refs/heads/master @ [b065097dfc3353bea82cea588ce337d4a7c435c6](https://github.com/cockroachdb/cockroach/commits/b065097dfc3353bea82cea588ce337d4a7c435c6):


```
          ----
          {s4:*,s5,s6}:1
          {s7:*,s8,s9}:1
    datadriven_simulation_test.go:199: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2335/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/asim/tests/tests_test_/tests_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/asim/tests/testdata/non_rand/example_skewed_cpu_even_ranges_mma.txt:28:
          gen_load
          ----
    datadriven_simulation_test.go:199: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2335/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/asim/tests/tests_test_/tests_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/asim/tests/testdata/non_rand/example_skewed_cpu_even_ranges_mma.txt:31:
          setting
          ----
    datadriven_simulation_test.go:461: running eval for example_skewed_cpu_even_ranges_mma
    datadriven_simulation_test.go:199: 
          /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2335/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/asim/tests/tests_test_/tests_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/asim/tests/testdata/non_rand/example_skewed_cpu_even_ranges_mma.txt:37:
          eval
        output didn't match expected:
          @@ -4,19 +4,19 @@
           replicas#1: first: [s1=36, s2=36, s3=36, s4=36, s5=36, s6=36, s7=36, s8=36, s9=36] (stddev=0.00, mean=36.00, sum=324)
           replicas#1: last:  [s1=36, s2=37, s3=33, s4=37, s5=36, s6=36, s7=36, s8=35, s9=38] (stddev=1.33, mean=36.00, sum=324)
           write_bytes_per_second#1: last:  [s1=5619, s2=5639, s3=5275, s4=6269, s5=6137, s6=6150, s7=6187, s8=5861, s9=6290] (stddev=336.10, mean=5936.33, sum=53427)
           artifacts[sma-only]: 6f86408eff3edb14
           cpu#1: last:  [s1=572909386, s2=573979147, s3=572126524, s4=557253107, s5=557048324, s6=558469521, s7=571735501, s8=555522023, s9=554214626] (stddev=8138152.97, mean=563695351.00, sum=5073258159)
          -leases#1: first: [s1=36, s2=0, s3=0, s4=36, s5=0, s6=0, s7=36, s8=0, s9=0] (stddev=16.97, mean=12.00, sum=108)
          -leases#1: last:  [s1=4, s2=4, s3=4, s4=40, s5=4, s6=4, s7=40, s8=4, s9=4] (stddev=14.97, mean=12.00, sum=108)
          -replicas#1: first: [s1=36, s2=36, s3=36, s4=36, s5=36, s6=36, s7=36, s8=36, s9=36] (stddev=0.00, mean=36.00, sum=324)
          -replicas#1: last:  [s1=28, s2=28, s3=28, s4=40, s5=40, s6=40, s7=40, s8=40, s9=40] (stddev=5.66, mean=36.00, sum=324)
          -write_bytes_per_second#1: last:  [s1=3898, s2=3891, s3=3878, s4=6939, s5=6939, s6=6929, s7=6990, s8=6964, s9=6978] (stddev=1446.16, mean=5934.00, sum=53406)
          -artifacts[mma-only]: 1d73e264a60412ba
          -cpu#1: last:  [s1=572415460, s2=566578888, s3=569096957, s4=556745916, s5=570328363, s6=558184663, s7=560070762, s8=557855218, s9=558141451] (stddev=5893908.39, mean=563268630.89, sum=5069417678)
          -leases#1: first: [s1=36, s2=0, s3=0, s4=36, s5=0, s6=0, s7=36, s8=0, s9=0] (stddev=16.97, mean=12.00, sum=108)
          -leases#1: last:  [s1=7, s2=10, s3=8, s4=17, s5=13, s6=13, s7=16, s8=12, s9=12] (stddev=3.13, mean=12.00, sum=108)
          -replicas#1: first: [s1=36, s2=36, s3=36, s4=36, s5=36, s6=36, s7=36, s8=36, s9=36] (stddev=0.00, mean=36.00, sum=324)
          -replicas#1: last:  [s1=36, s2=34, s3=36, s4=37, s5=36, s6=36, s7=37, s8=36, s9=36] (stddev=0.82, mean=36.00, sum=324)
          -write_bytes_per_second#1: last:  [s1=5194, s2=5256, s3=5415, s4=6318, s5=6216, s6=6171, s7=6283, s8=6298, s9=6257] (stddev=461.69, mean=5934.22, sum=53408)
          -artifacts[both]: 4fab04c2e6adb6a5
          +leases#1: first: [s1=36, s2=0, s3=0, s4=72, s5=0, s6=0, s7=0, s8=0, s9=0] (stddev=24.00, mean=12.00, sum=108)
          +leases#1: last:  [s1=4, s2=4, s3=4, s4=76, s5=4, s6=4, s7=4, s8=4, s9=4] (stddev=22.63, mean=12.00, sum=108)
          +replicas#1: first: [s1=36, s2=36, s3=36, s4=72, s5=72, s6=72, s7=0, s8=0, s9=0] (stddev=29.39, mean=36.00, sum=324)
          +replicas#1: last:  [s1=28, s2=28, s3=28, s4=76, s5=76, s6=76, s7=4, s8=4, s9=4] (stddev=29.93, mean=36.00, sum=324)
          +write_bytes_per_second#1: last:  [s1=3898, s2=3891, s3=3878, s4=13362, s5=13362, s6=13352, s7=567, s8=541, s9=555] (stddev=5423.67, mean=5934.00, sum=53406)
          +artifacts[mma-only]: b49ca7cf1d2fa63f
          +cpu#1: last:  [s1=706370034, s2=563590134, s3=564287545, s4=556031498, s5=556687750, s6=574261650, s7=559888693, s8=562039766, s9=421135179] (stddev=67418771.72, mean=562699138.78, sum=5064292249)
          +leases#1: first: [s1=36, s2=0, s3=0, s4=72, s5=0, s6=0, s7=0, s8=0, s9=0] (stddev=24.00, mean=12.00, sum=108)
          +leases#1: last:  [s1=7, s2=11, s3=10, s4=17, s5=16, s6=12, s7=12, s8=9, s9=14] (stddev=3.06, mean=12.00, sum=108)
          +replicas#1: first: [s1=36, s2=36, s3=36, s4=72, s5=72, s6=72, s7=0, s8=0, s9=0] (stddev=29.39, mean=36.00, sum=324)
          +replicas#1: last:  [s1=36, s2=35, s3=36, s4=36, s5=35, s6=37, s7=36, s8=36, s9=37] (stddev=0.67, mean=36.00, sum=324)
          +write_bytes_per_second#1: last:  [s1=5151, s2=5813, s3=5663, s4=6148, s5=6043, s6=6286, s7=6089, s8=5955, s9=6234] (stddev=332.86, mean=5931.33, sum=53382)
          +artifacts[both]: 75673788ebbf338b
           
--- FAIL: TestDataDriven/example_skewed_cpu_even_ranges_mma (27.65s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv-triage @cockroachdb/kv-triage
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDataDriven.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-53149

kv/kvserver/asim/tests.TestDataDriven [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_NightlyCockroachExtraStress/20288774?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_NightlyCockroachExtraStress/20288774?buildTab=artifacts#/) on refs/heads/master @ [a4ad94ae1379daa20b467c89d09de27ba8e2356d](https://github.com/cockroachdb/cockroach/commits/a4ad94ae1379daa20b467c89d09de27ba8e2356d):


```
          ----
          {s4:*,s5,s6}:1
          {s7:*,s8,s9}:1
    datadriven_simulation_test.go:199: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2335/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/asim/tests/tests_test_/tests_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/asim/tests/testdata/non_rand/example_skewed_cpu_even_ranges_mma.txt:28:
          gen_load
          ----
    datadriven_simulation_test.go:199: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2335/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/asim/tests/tests_test_/tests_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/asim/tests/testdata/non_rand/example_skewed_cpu_even_ranges_mma.txt:31:
          setting
          ----
    datadriven_simulation_test.go:461: running eval for example_skewed_cpu_even_ranges_mma
    datadriven_simulation_test.go:199: 
          /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2335/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/asim/tests/tests_test_/tests_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/asim/tests/testdata/non_rand/example_skewed_cpu_even_ranges_mma.txt:37:
          eval
        output didn't match expected:
          @@ -4,19 +4,19 @@
           replicas#1: first: [s1=36, s2=36, s3=36, s4=36, s5=36, s6=36, s7=36, s8=36, s9=36] (stddev=0.00, mean=36.00, sum=324)
           replicas#1: last:  [s1=36, s2=37, s3=33, s4=37, s5=36, s6=36, s7=36, s8=35, s9=38] (stddev=1.33, mean=36.00, sum=324)
           write_bytes_per_second#1: last:  [s1=5619, s2=5639, s3=5275, s4=6269, s5=6137, s6=6150, s7=6187, s8=5861, s9=6290] (stddev=336.10, mean=5936.33, sum=53427)
           artifacts[sma-only]: 6f86408eff3edb14
           cpu#1: last:  [s1=572909386, s2=573979147, s3=572126524, s4=557253107, s5=557048324, s6=558469521, s7=571735501, s8=555522023, s9=554214626] (stddev=8138152.97, mean=563695351.00, sum=5073258159)
          -leases#1: first: [s1=36, s2=0, s3=0, s4=36, s5=0, s6=0, s7=36, s8=0, s9=0] (stddev=16.97, mean=12.00, sum=108)
          -leases#1: last:  [s1=4, s2=4, s3=4, s4=40, s5=4, s6=4, s7=40, s8=4, s9=4] (stddev=14.97, mean=12.00, sum=108)
          -replicas#1: first: [s1=36, s2=36, s3=36, s4=36, s5=36, s6=36, s7=36, s8=36, s9=36] (stddev=0.00, mean=36.00, sum=324)
          -replicas#1: last:  [s1=28, s2=28, s3=28, s4=40, s5=40, s6=40, s7=40, s8=40, s9=40] (stddev=5.66, mean=36.00, sum=324)
          -write_bytes_per_second#1: last:  [s1=3898, s2=3891, s3=3878, s4=6939, s5=6939, s6=6929, s7=6990, s8=6964, s9=6978] (stddev=1446.16, mean=5934.00, sum=53406)
          -artifacts[mma-only]: 1d73e264a60412ba
          -cpu#1: last:  [s1=572415460, s2=566578888, s3=569096957, s4=556745916, s5=570328363, s6=558184663, s7=560070762, s8=557855218, s9=558141451] (stddev=5893908.39, mean=563268630.89, sum=5069417678)
          -leases#1: first: [s1=36, s2=0, s3=0, s4=36, s5=0, s6=0, s7=36, s8=0, s9=0] (stddev=16.97, mean=12.00, sum=108)
          -leases#1: last:  [s1=7, s2=10, s3=8, s4=17, s5=13, s6=13, s7=16, s8=12, s9=12] (stddev=3.13, mean=12.00, sum=108)
          -replicas#1: first: [s1=36, s2=36, s3=36, s4=36, s5=36, s6=36, s7=36, s8=36, s9=36] (stddev=0.00, mean=36.00, sum=324)
          -replicas#1: last:  [s1=36, s2=34, s3=36, s4=37, s5=36, s6=36, s7=37, s8=36, s9=36] (stddev=0.82, mean=36.00, sum=324)
          -write_bytes_per_second#1: last:  [s1=5194, s2=5256, s3=5415, s4=6318, s5=6216, s6=6171, s7=6283, s8=6298, s9=6257] (stddev=461.69, mean=5934.22, sum=53408)
          -artifacts[both]: 4fab04c2e6adb6a5
          +leases#1: first: [s1=36, s2=0, s3=0, s4=72, s5=0, s6=0, s7=0, s8=0, s9=0] (stddev=24.00, mean=12.00, sum=108)
          +leases#1: last:  [s1=4, s2=4, s3=4, s4=76, s5=4, s6=4, s7=4, s8=4, s9=4] (stddev=22.63, mean=12.00, sum=108)
          +replicas#1: first: [s1=36, s2=36, s3=36, s4=72, s5=72, s6=72, s7=0, s8=0, s9=0] (stddev=29.39, mean=36.00, sum=324)
          +replicas#1: last:  [s1=28, s2=28, s3=28, s4=76, s5=76, s6=76, s7=4, s8=4, s9=4] (stddev=29.93, mean=36.00, sum=324)
          +write_bytes_per_second#1: last:  [s1=3898, s2=3891, s3=3878, s4=13362, s5=13362, s6=13352, s7=567, s8=541, s9=555] (stddev=5423.67, mean=5934.00, sum=53406)
          +artifacts[mma-only]: b49ca7cf1d2fa63f
          +cpu#1: last:  [s1=706370034, s2=563590134, s3=564287545, s4=556031498, s5=556687750, s6=574261650, s7=559888693, s8=562039766, s9=421135179] (stddev=67418771.72, mean=562699138.78, sum=5064292249)
          +leases#1: first: [s1=36, s2=0, s3=0, s4=72, s5=0, s6=0, s7=0, s8=0, s9=0] (stddev=24.00, mean=12.00, sum=108)
          +leases#1: last:  [s1=7, s2=11, s3=10, s4=17, s5=16, s6=12, s7=12, s8=9, s9=14] (stddev=3.06, mean=12.00, sum=108)
          +replicas#1: first: [s1=36, s2=36, s3=36, s4=72, s5=72, s6=72, s7=0, s8=0, s9=0] (stddev=29.39, mean=36.00, sum=324)
          +replicas#1: last:  [s1=36, s2=35, s3=36, s4=36, s5=35, s6=37, s7=36, s8=36, s9=37] (stddev=0.67, mean=36.00, sum=324)
          +write_bytes_per_second#1: last:  [s1=5151, s2=5813, s3=5663, s4=6148, s5=6043, s6=6286, s7=6089, s8=5955, s9=6234] (stddev=332.86, mean=5931.33, sum=53382)
          +artifacts[both]: 75673788ebbf338b
           
--- FAIL: TestDataDriven/example_skewed_cpu_even_ranges_mma (23.17s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDataDriven.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

kv/kvserver/asim/tests.TestDataDriven [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_NightlyCockroachExtraStress/20289631?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_NightlyCockroachExtraStress/20289631?buildTab=artifacts#/) on refs/heads/master @ [ca0d4076c381609d6487040141c87e36a68b211f](https://github.com/cockroachdb/cockroach/commits/ca0d4076c381609d6487040141c87e36a68b211f):


```
          ----
          {s4:*,s5,s6}:1
          {s7:*,s8,s9}:1
    datadriven_simulation_test.go:199: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2335/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/asim/tests/tests_test_/tests_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/asim/tests/testdata/non_rand/example_skewed_cpu_even_ranges_mma.txt:28:
          gen_load
          ----
    datadriven_simulation_test.go:199: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2335/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/asim/tests/tests_test_/tests_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/asim/tests/testdata/non_rand/example_skewed_cpu_even_ranges_mma.txt:31:
          setting
          ----
    datadriven_simulation_test.go:461: running eval for example_skewed_cpu_even_ranges_mma
    datadriven_simulation_test.go:199: 
          /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2335/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/asim/tests/tests_test_/tests_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/asim/tests/testdata/non_rand/example_skewed_cpu_even_ranges_mma.txt:37:
          eval
        output didn't match expected:
          @@ -4,19 +4,19 @@
           replicas#1: first: [s1=36, s2=36, s3=36, s4=36, s5=36, s6=36, s7=36, s8=36, s9=36] (stddev=0.00, mean=36.00, sum=324)
           replicas#1: last:  [s1=36, s2=37, s3=33, s4=37, s5=36, s6=36, s7=36, s8=35, s9=38] (stddev=1.33, mean=36.00, sum=324)
           write_bytes_per_second#1: last:  [s1=5619, s2=5639, s3=5275, s4=6269, s5=6137, s6=6150, s7=6187, s8=5861, s9=6290] (stddev=336.10, mean=5936.33, sum=53427)
           artifacts[sma-only]: 6f86408eff3edb14
           cpu#1: last:  [s1=572909386, s2=573979147, s3=572126524, s4=557253107, s5=557048324, s6=558469521, s7=571735501, s8=555522023, s9=554214626] (stddev=8138152.97, mean=563695351.00, sum=5073258159)
          -leases#1: first: [s1=36, s2=0, s3=0, s4=36, s5=0, s6=0, s7=36, s8=0, s9=0] (stddev=16.97, mean=12.00, sum=108)
          -leases#1: last:  [s1=4, s2=4, s3=4, s4=40, s5=4, s6=4, s7=40, s8=4, s9=4] (stddev=14.97, mean=12.00, sum=108)
          -replicas#1: first: [s1=36, s2=36, s3=36, s4=36, s5=36, s6=36, s7=36, s8=36, s9=36] (stddev=0.00, mean=36.00, sum=324)
          -replicas#1: last:  [s1=28, s2=28, s3=28, s4=40, s5=40, s6=40, s7=40, s8=40, s9=40] (stddev=5.66, mean=36.00, sum=324)
          -write_bytes_per_second#1: last:  [s1=3898, s2=3891, s3=3878, s4=6939, s5=6939, s6=6929, s7=6990, s8=6964, s9=6978] (stddev=1446.16, mean=5934.00, sum=53406)
          -artifacts[mma-only]: 1d73e264a60412ba
          -cpu#1: last:  [s1=572415460, s2=566578888, s3=569096957, s4=556745916, s5=570328363, s6=558184663, s7=560070762, s8=557855218, s9=558141451] (stddev=5893908.39, mean=563268630.89, sum=5069417678)
          -leases#1: first: [s1=36, s2=0, s3=0, s4=36, s5=0, s6=0, s7=36, s8=0, s9=0] (stddev=16.97, mean=12.00, sum=108)
          -leases#1: last:  [s1=7, s2=10, s3=8, s4=17, s5=13, s6=13, s7=16, s8=12, s9=12] (stddev=3.13, mean=12.00, sum=108)
          -replicas#1: first: [s1=36, s2=36, s3=36, s4=36, s5=36, s6=36, s7=36, s8=36, s9=36] (stddev=0.00, mean=36.00, sum=324)
          -replicas#1: last:  [s1=36, s2=34, s3=36, s4=37, s5=36, s6=36, s7=37, s8=36, s9=36] (stddev=0.82, mean=36.00, sum=324)
          -write_bytes_per_second#1: last:  [s1=5194, s2=5256, s3=5415, s4=6318, s5=6216, s6=6171, s7=6283, s8=6298, s9=6257] (stddev=461.69, mean=5934.22, sum=53408)
          -artifacts[both]: 4fab04c2e6adb6a5
          +leases#1: first: [s1=36, s2=0, s3=0, s4=72, s5=0, s6=0, s7=0, s8=0, s9=0] (stddev=24.00, mean=12.00, sum=108)
          +leases#1: last:  [s1=4, s2=4, s3=4, s4=76, s5=4, s6=4, s7=4, s8=4, s9=4] (stddev=22.63, mean=12.00, sum=108)
          +replicas#1: first: [s1=36, s2=36, s3=36, s4=72, s5=72, s6=72, s7=0, s8=0, s9=0] (stddev=29.39, mean=36.00, sum=324)
          +replicas#1: last:  [s1=28, s2=28, s3=28, s4=76, s5=76, s6=76, s7=4, s8=4, s9=4] (stddev=29.93, mean=36.00, sum=324)
          +write_bytes_per_second#1: last:  [s1=3898, s2=3891, s3=3878, s4=13362, s5=13362, s6=13352, s7=567, s8=541, s9=555] (stddev=5423.67, mean=5934.00, sum=53406)
          +artifacts[mma-only]: b49ca7cf1d2fa63f
          +cpu#1: last:  [s1=706370034, s2=563590134, s3=564287545, s4=556031498, s5=556687750, s6=574261650, s7=559888693, s8=562039766, s9=421135179] (stddev=67418771.72, mean=562699138.78, sum=5064292249)
          +leases#1: first: [s1=36, s2=0, s3=0, s4=72, s5=0, s6=0, s7=0, s8=0, s9=0] (stddev=24.00, mean=12.00, sum=108)
          +leases#1: last:  [s1=7, s2=11, s3=10, s4=17, s5=16, s6=12, s7=12, s8=9, s9=14] (stddev=3.06, mean=12.00, sum=108)
          +replicas#1: first: [s1=36, s2=36, s3=36, s4=72, s5=72, s6=72, s7=0, s8=0, s9=0] (stddev=29.39, mean=36.00, sum=324)
          +replicas#1: last:  [s1=36, s2=35, s3=36, s4=36, s5=35, s6=37, s7=36, s8=36, s9=37] (stddev=0.67, mean=36.00, sum=324)
          +write_bytes_per_second#1: last:  [s1=5151, s2=5813, s3=5663, s4=6148, s5=6043, s6=6286, s7=6089, s8=5955, s9=6234] (stddev=332.86, mean=5931.33, sum=53382)
          +artifacts[both]: 75673788ebbf338b
           
--- FAIL: TestDataDriven/example_skewed_cpu_even_ranges_mma (26.36s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDataDriven.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

kv/kvserver/asim/tests.TestDataDriven [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_NightlyCockroachExtraStress/20294761?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_NightlyCockroachExtraStress/20294761?buildTab=artifacts#/) on refs/heads/master @ [e670091c79e060ff0e84b2c89c6b82aebba85a08](https://github.com/cockroachdb/cockroach/commits/e670091c79e060ff0e84b2c89c6b82aebba85a08):


```
          ----
          {s4:*,s5,s6}:1
          {s7:*,s8,s9}:1
    datadriven_simulation_test.go:199: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2164/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/asim/tests/tests_test_/tests_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/asim/tests/testdata/non_rand/example_skewed_cpu_even_ranges_mma.txt:28:
          gen_load
          ----
    datadriven_simulation_test.go:199: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2164/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/asim/tests/tests_test_/tests_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/asim/tests/testdata/non_rand/example_skewed_cpu_even_ranges_mma.txt:31:
          setting
          ----
    datadriven_simulation_test.go:461: running eval for example_skewed_cpu_even_ranges_mma
    datadriven_simulation_test.go:199: 
          /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2164/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/asim/tests/tests_test_/tests_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/asim/tests/testdata/non_rand/example_skewed_cpu_even_ranges_mma.txt:37:
          eval
        output didn't match expected:
          @@ -4,19 +4,19 @@
           replicas#1: first: [s1=36, s2=36, s3=36, s4=36, s5=36, s6=36, s7=36, s8=36, s9=36] (stddev=0.00, mean=36.00, sum=324)
           replicas#1: last:  [s1=36, s2=37, s3=33, s4=37, s5=36, s6=36, s7=36, s8=35, s9=38] (stddev=1.33, mean=36.00, sum=324)
           write_bytes_per_second#1: last:  [s1=5619, s2=5639, s3=5275, s4=6269, s5=6137, s6=6150, s7=6187, s8=5861, s9=6290] (stddev=336.10, mean=5936.33, sum=53427)
           artifacts[sma-only]: 6f86408eff3edb14
           cpu#1: last:  [s1=572909386, s2=573979147, s3=572126524, s4=557253107, s5=557048324, s6=558469521, s7=571735501, s8=555522023, s9=554214626] (stddev=8138152.97, mean=563695351.00, sum=5073258159)
          -leases#1: first: [s1=36, s2=0, s3=0, s4=36, s5=0, s6=0, s7=36, s8=0, s9=0] (stddev=16.97, mean=12.00, sum=108)
          -leases#1: last:  [s1=4, s2=4, s3=4, s4=40, s5=4, s6=4, s7=40, s8=4, s9=4] (stddev=14.97, mean=12.00, sum=108)
          -replicas#1: first: [s1=36, s2=36, s3=36, s4=36, s5=36, s6=36, s7=36, s8=36, s9=36] (stddev=0.00, mean=36.00, sum=324)
          -replicas#1: last:  [s1=28, s2=28, s3=28, s4=40, s5=40, s6=40, s7=40, s8=40, s9=40] (stddev=5.66, mean=36.00, sum=324)
          -write_bytes_per_second#1: last:  [s1=3898, s2=3891, s3=3878, s4=6939, s5=6939, s6=6929, s7=6990, s8=6964, s9=6978] (stddev=1446.16, mean=5934.00, sum=53406)
          -artifacts[mma-only]: 1d73e264a60412ba
          -cpu#1: last:  [s1=572415460, s2=566578888, s3=569096957, s4=556745916, s5=570328363, s6=558184663, s7=560070762, s8=557855218, s9=558141451] (stddev=5893908.39, mean=563268630.89, sum=5069417678)
          -leases#1: first: [s1=36, s2=0, s3=0, s4=36, s5=0, s6=0, s7=36, s8=0, s9=0] (stddev=16.97, mean=12.00, sum=108)
          -leases#1: last:  [s1=7, s2=10, s3=8, s4=17, s5=13, s6=13, s7=16, s8=12, s9=12] (stddev=3.13, mean=12.00, sum=108)
          -replicas#1: first: [s1=36, s2=36, s3=36, s4=36, s5=36, s6=36, s7=36, s8=36, s9=36] (stddev=0.00, mean=36.00, sum=324)
          -replicas#1: last:  [s1=36, s2=34, s3=36, s4=37, s5=36, s6=36, s7=37, s8=36, s9=36] (stddev=0.82, mean=36.00, sum=324)
          -write_bytes_per_second#1: last:  [s1=5194, s2=5256, s3=5415, s4=6318, s5=6216, s6=6171, s7=6283, s8=6298, s9=6257] (stddev=461.69, mean=5934.22, sum=53408)
          -artifacts[both]: 4fab04c2e6adb6a5
          +leases#1: first: [s1=36, s2=0, s3=0, s4=72, s5=0, s6=0, s7=0, s8=0, s9=0] (stddev=24.00, mean=12.00, sum=108)
          +leases#1: last:  [s1=4, s2=4, s3=4, s4=76, s5=4, s6=4, s7=4, s8=4, s9=4] (stddev=22.63, mean=12.00, sum=108)
          +replicas#1: first: [s1=36, s2=36, s3=36, s4=72, s5=72, s6=72, s7=0, s8=0, s9=0] (stddev=29.39, mean=36.00, sum=324)
          +replicas#1: last:  [s1=28, s2=28, s3=28, s4=76, s5=76, s6=76, s7=4, s8=4, s9=4] (stddev=29.93, mean=36.00, sum=324)
          +write_bytes_per_second#1: last:  [s1=3898, s2=3891, s3=3878, s4=13362, s5=13362, s6=13352, s7=567, s8=541, s9=555] (stddev=5423.67, mean=5934.00, sum=53406)
          +artifacts[mma-only]: b49ca7cf1d2fa63f
          +cpu#1: last:  [s1=706370034, s2=563590134, s3=564287545, s4=556031498, s5=556687750, s6=574261650, s7=559888693, s8=562039766, s9=421135179] (stddev=67418771.72, mean=562699138.78, sum=5064292249)
          +leases#1: first: [s1=36, s2=0, s3=0, s4=72, s5=0, s6=0, s7=0, s8=0, s9=0] (stddev=24.00, mean=12.00, sum=108)
          +leases#1: last:  [s1=7, s2=11, s3=10, s4=17, s5=16, s6=12, s7=12, s8=9, s9=14] (stddev=3.06, mean=12.00, sum=108)
          +replicas#1: first: [s1=36, s2=36, s3=36, s4=72, s5=72, s6=72, s7=0, s8=0, s9=0] (stddev=29.39, mean=36.00, sum=324)
          +replicas#1: last:  [s1=36, s2=35, s3=36, s4=36, s5=35, s6=37, s7=36, s8=36, s9=37] (stddev=0.67, mean=36.00, sum=324)
          +write_bytes_per_second#1: last:  [s1=5151, s2=5813, s3=5663, s4=6148, s5=6043, s6=6286, s7=6089, s8=5955, s9=6234] (stddev=332.86, mean=5931.33, sum=53382)
          +artifacts[both]: 75673788ebbf338b
           
--- FAIL: TestDataDriven/example_skewed_cpu_even_ranges_mma (28.32s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDataDriven.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

kv/kvserver/asim/tests.TestDataDriven [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_NightlyCockroachExtraStress/20296037?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_NightlyCockroachExtraStress/20296037?buildTab=artifacts#/) on refs/heads/master @ [a324bdd9cbccf7be4172a3510d32071bc45bf98b](https://github.com/cockroachdb/cockroach/commits/a324bdd9cbccf7be4172a3510d32071bc45bf98b):


```
          eval
        
        expected:
          artifacts[default]: 48fc2e7a0c77951b
        
        found:
          artifacts[default]: fb6d3b238ce25bbb
--- FAIL: TestDataDriven/example_zone_config (5.27s)
=== RUN   TestDataDriven/mma_high_cpu_25nodes
=== PAUSE TestDataDriven/mma_high_cpu_25nodes
=== CONT  TestDataDriven/mma_high_cpu_25nodes
    datadriven_simulation_test.go:216: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/73/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/asim/tests/tests_test_/tests_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/asim/tests/testdata/non_rand/mma_high_cpu_25nodes.txt:1:
          skip_under_ci
          ----
    datadriven_simulation_test.go:216: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/73/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/asim/tests/tests_test_/tests_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/asim/tests/testdata/non_rand/mma_high_cpu_25nodes.txt:4:
          gen_cluster
          ----
    datadriven_simulation_test.go:216: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/73/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/asim/tests/tests_test_/tests_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/asim/tests/testdata/non_rand/mma_high_cpu_25nodes.txt:10:
          setting
          ----
    datadriven_simulation_test.go:216: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/73/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/asim/tests/tests_test_/tests_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/asim/tests/testdata/non_rand/mma_high_cpu_25nodes.txt:14:
          gen_ranges
          ----
    datadriven_simulation_test.go:216: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/73/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/asim/tests/tests_test_/tests_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/asim/tests/testdata/non_rand/mma_high_cpu_25nodes.txt:17:
          gen_load
          ----
    datadriven_simulation_test.go:216: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/73/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/asim/tests/tests_test_/tests_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/asim/tests/testdata/non_rand/mma_high_cpu_25nodes.txt:22:
          gen_ranges
          ----
    datadriven_simulation_test.go:216: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/73/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/asim/tests/tests_test_/tests_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/asim/tests/testdata/non_rand/mma_high_cpu_25nodes.txt:25:
          gen_load
          ----
    datadriven_simulation_test.go:432: running eval for mma_high_cpu_25nodes.txt
    datadriven_simulation_test.go:216: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/73/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/asim/tests/tests_test_/tests_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/asim/tests/testdata/non_rand/mma_high_cpu_25nodes.txt:28:
          eval
        
        expected:
          artifacts[default]: c51304df5082c6ed
        
        found:
          artifacts[default]: da081062c0a0876e
--- FAIL: TestDataDriven/mma_high_cpu_25nodes (53.80s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDataDriven.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

kv/kvserver/asim/tests.TestDataDriven [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_NightlyCockroachExtraStress/20296039?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_NightlyCockroachExtraStress/20296039?buildTab=artifacts#/) on refs/heads/master @ [9a2728159aca5a90da123ecc85909f5d619ead60](https://github.com/cockroachdb/cockroach/commits/9a2728159aca5a90da123ecc85909f5d619ead60):


```
          +leases#1: last:  [s1=7, s2=11, s3=10, s4=16, s5=17, s6=12, s7=12, s8=8, s9=15] (stddev=3.27, mean=12.00, sum=108)
          +replicas#1: first: [s1=36, s2=36, s3=36, s4=72, s5=72, s6=72, s7=0, s8=0, s9=0] (stddev=29.39, mean=36.00, sum=324)
          +replicas#1: last:  [s1=36, s2=35, s3=36, s4=35, s5=36, s6=37, s7=36, s8=36, s9=37] (stddev=0.67, mean=36.00, sum=324)
          +write_bytes_per_second#1: last:  [s1=5164, s2=5832, s3=5697, s4=6012, s5=6183, s6=6265, s7=6162, s8=5963, s9=6285] (stddev=335.06, mean=5951.44, sum=53563)
          +artifacts[both]: de9625c5922d24b5
           
--- FAIL: TestDataDriven/example_skewed_cpu_even_ranges_mma (19.04s)
=== RUN   TestDataDriven/example_zone_config
=== PAUSE TestDataDriven/example_zone_config
=== CONT  TestDataDriven/example_zone_config
    datadriven_simulation_test.go:200: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/10/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/asim/tests/tests_test_/tests_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/asim/tests/testdata/non_rand/example_zone_config.txt:1:
          skip_under_ci
          ----
    datadriven_simulation_test.go:200: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/10/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/asim/tests/tests_test_/tests_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/asim/tests/testdata/non_rand/example_zone_config.txt:10:
          load_cluster
          ----
    datadriven_simulation_test.go:200: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/10/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/asim/tests/tests_test_/tests_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/asim/tests/testdata/non_rand/example_zone_config.txt:14:
          set_span_config
          [0,9999999999): num_replicas=3 constraints={'+region=US_East'}
          ----
    datadriven_simulation_test.go:200: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/10/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/asim/tests/tests_test_/tests_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/asim/tests/testdata/non_rand/example_zone_config.txt:18:
          assertion
          ----
    datadriven_simulation_test.go:200: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/10/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/asim/tests/tests_test_/tests_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/asim/tests/testdata/non_rand/example_zone_config.txt:21:
          gen_ranges
          ----
    datadriven_simulation_test.go:200: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/10/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/asim/tests/tests_test_/tests_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/asim/tests/testdata/non_rand/example_zone_config.txt:25:
          assertion
          ----
    datadriven_simulation_test.go:457: running eval for example_zone_config
    datadriven_simulation_test.go:200: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/10/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/asim/tests/tests_test_/tests_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/asim/tests/testdata/non_rand/example_zone_config.txt:28:
          eval
        
        expected:
          replicas#1: first: [s1=17, s2=17, s3=17, s4=17, s5=17, s6=16, s7=16, s8=17, s9=17, s10=17, s11=17, s12=17, s13=16, s14=16, s15=16, s16=16, s17=17, s18=17, s19=17, s20=16, s21=17, s22=16, s23=17, s24=17, s25=17, s26=17, s27=17, s28=16, s29=17, s30=16, s31=17, s32=17, s33=16, s34=16, s35=17, s36=17] (stddev=0.47, mean=16.67, sum=600)
          replicas#1: last:  [s1=51, s2=50, s3=52, s4=51, s5=49, s6=49, s7=50, s8=49, s9=49, s10=50, s11=48, s12=52, s13=0, s14=0, s15=0, s16=0, s17=0, s18=0, s19=0, s20=0, s21=0, s22=0, s23=0, s24=0, s25=0, s26=0, s27=0, s28=0, s29=0, s30=0, s31=0, s32=0, s33=0, s34=0, s35=0, s36=0] (stddev=23.58, mean=16.67, sum=600)
          artifacts[default]: 48fc2e7a0c77951b
        
        found:
          replicas#1: first: [s1=17, s2=17, s3=17, s4=17, s5=17, s6=16, s7=17, s8=16, s9=16, s10=17, s11=17, s12=16, s13=17, s14=17, s15=17, s16=16, s17=17, s18=17, s19=17, s20=17, s21=16, s22=16, s23=17, s24=17, s25=17, s26=17, s27=17, s28=17, s29=16, s30=17, s31=16, s32=17, s33=16, s34=16, s35=17, s36=16] (stddev=0.47, mean=16.67, sum=600)
          replicas#1: last:  [s1=50, s2=50, s3=50, s4=49, s5=50, s6=50, s7=51, s8=50, s9=51, s10=50, s11=49, s12=50, s13=0, s14=0, s15=0, s16=0, s17=0, s18=0, s19=0, s20=0, s21=0, s22=0, s23=0, s24=0, s25=0, s26=0, s27=0, s28=0, s29=0, s30=0, s31=0, s32=0, s33=0, s34=0, s35=0, s36=0] (stddev=23.57, mean=16.67, sum=600)
          artifacts[default]: fb6d3b238ce25bbb
--- FAIL: TestDataDriven/example_zone_config (5.34s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDataDriven.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

kv/kvserver/asim/tests.TestDataDriven [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_NightlyCockroachExtraStress/20296147?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_NightlyCockroachExtraStress/20296147?buildTab=artifacts#/) on refs/heads/master @ [7a9149e4dc78867cfa054baeded1f3e8f1fe5954](https://github.com/cockroachdb/cockroach/commits/7a9149e4dc78867cfa054baeded1f3e8f1fe5954):


```
=== PAUSE TestDataDriven/mma_high_cpu_25nodes
=== CONT  TestDataDriven/mma_high_cpu_25nodes
    datadriven_simulation_test.go:200: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/10/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/asim/tests/tests_test_/tests_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/asim/tests/testdata/non_rand/mma_high_cpu_25nodes.txt:1:
          skip_under_ci
          ----
    datadriven_simulation_test.go:200: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/10/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/asim/tests/tests_test_/tests_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/asim/tests/testdata/non_rand/mma_high_cpu_25nodes.txt:4:
          gen_cluster
          ----
    datadriven_simulation_test.go:200: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/10/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/asim/tests/tests_test_/tests_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/asim/tests/testdata/non_rand/mma_high_cpu_25nodes.txt:7:
          setting
          ----
    datadriven_simulation_test.go:200: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/10/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/asim/tests/tests_test_/tests_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/asim/tests/testdata/non_rand/mma_high_cpu_25nodes.txt:11:
          gen_ranges
          ----
    datadriven_simulation_test.go:200: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/10/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/asim/tests/tests_test_/tests_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/asim/tests/testdata/non_rand/mma_high_cpu_25nodes.txt:14:
          gen_load
          ----
    datadriven_simulation_test.go:200: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/10/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/asim/tests/tests_test_/tests_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/asim/tests/testdata/non_rand/mma_high_cpu_25nodes.txt:19:
          gen_ranges
          ----
    datadriven_simulation_test.go:200: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/10/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/asim/tests/tests_test_/tests_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/asim/tests/testdata/non_rand/mma_high_cpu_25nodes.txt:22:
          gen_load
          ----
    datadriven_simulation_test.go:457: running eval for mma_high_cpu_25nodes
    datadriven_simulation_test.go:200: 
          /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/10/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/asim/tests/tests_test_/tests_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/asim/tests/testdata/non_rand/mma_high_cpu_25nodes.txt:25:
          eval
        output didn't match expected:
          @@ -1,8 +1,8 @@
          -cpu#1: last:  [s1=100080327, s2=101363144, s3=70767224, s4=60649264, s5=60712788, s6=61798, s7=60732184, s8=60583825, s9=60431631, s10=60604936, s11=60635876, s12=60666240, s13=60711555, s14=60715536, s15=60704931, s16=60603244, s17=60498137, s18=60757306, s19=60625361, s20=60472051, s21=60628443, s22=60830571, s23=60797391, s24=60672637, s25=60668417] (stddev=16673880.98, mean=61838992.68, sum=1545974817)
          -leases#1: first: [s1=51, s2=2, s3=2, s4=2, s5=2, s6=2, s7=2, s8=1, s9=2, s10=2, s11=3, s12=1, s13=3, s14=2, s15=2, s16=2, s17=2, s18=2, s19=2, s20=2, s21=2, s22=2, s23=3, s24=3, s25=1] (stddev=9.61, mean=4.00, sum=100)
          -leases#1: last:  [s1=4, s2=5, s3=4, s4=4, s5=4, s6=2, s7=4, s8=3, s9=4, s10=4, s11=5, s12=3, s13=5, s14=4, s15=4, s16=4, s17=4, s18=4, s19=4, s20=4, s21=4, s22=4, s23=5, s24=5, s25=3] (stddev=0.69, mean=4.00, sum=100)
          +cpu#1: last:  [s1=100110376, s2=101363178, s3=70767240, s4=60619159, s5=60712903, s6=61812, s7=60762284, s8=60613711, s9=60431718, s10=60604975, s11=60635827, s12=60666150, s13=60651483, s14=60715674, s15=60735031, s16=60603145, s17=60498159, s18=60787310, s19=60625199, s20=60472014, s21=60628490, s22=60830573, s23=60767461, s24=60612569, s25=60698373] (stddev=16676733.67, mean=61838992.56, sum=1545974814)
          +leases#1: first: [s1=52, s2=2, s3=2, s4=1, s5=2, s6=2, s7=3, s8=2, s9=2, s10=2, s11=3, s12=1, s13=1, s14=2, s15=3, s16=2, s17=2, s18=3, s19=2, s20=2, s21=2, s22=2, s23=2, s24=1, s25=2] (stddev=9.81, mean=4.00, sum=100)
          +leases#1: last:  [s1=5, s2=5, s3=4, s4=3, s5=4, s6=2, s7=5, s8=4, s9=4, s10=4, s11=5, s12=3, s13=3, s14=4, s15=5, s16=4, s17=4, s18=5, s19=4, s20=4, s21=4, s22=4, s23=4, s24=3, s25=4] (stddev=0.75, mean=4.00, sum=100)
           replicas#1: first: [s1=56, s2=56, s3=56, s4=6, s5=6, s6=6, s7=6, s8=6, s9=6, s10=6, s11=6, s12=6, s13=6, s14=6, s15=6, s16=6, s17=6, s18=6, s19=6, s20=6, s21=6, s22=6, s23=6, s24=6, s25=6] (stddev=16.25, mean=12.00, sum=300)
           replicas#1: last:  [s1=40, s2=44, s3=42, s4=8, s5=8, s6=6, s7=8, s8=8, s9=8, s10=8, s11=8, s12=8, s13=8, s14=8, s15=8, s16=8, s17=8, s18=8, s19=8, s20=8, s21=8, s22=8, s23=8, s24=8, s25=8] (stddev=11.10, mean=12.00, sum=300)
          -write_bytes_per_second#1: last:  [s1=9514, s2=9566, s3=9510, s4=9031, s5=9020, s6=8984, s7=9004, s8=9027, s9=9050, s10=9034, s11=9012, s12=8981, s13=9043, s14=9042, s15=9071, s16=9054, s17=9029, s18=9030, s19=9020, s20=9005, s21=9031, s22=9035, s23=9038, s24=9054, s25=9050] (stddev=164.21, mean=9089.40, sum=227235)
          -artifacts[mma-only]: f12de3893f5fe99d
          +write_bytes_per_second#1: last:  [s1=9498, s2=9531, s3=9522, s4=8974, s5=9003, s6=9002, s7=9001, s8=9043, s9=9045, s10=9056, s11=8996, s12=9019, s13=9015, s14=9038, s15=9061, s16=9067, s17=9074, s18=9044, s19=9016, s20=9027, s21=9056, s22=9027, s23=9050, s24=9021, s25=9051] (stddev=159.76, mean=9089.48, sum=227237)
          +artifacts[mma-only]: 879f9bd640a13e1d
           
--- FAIL: TestDataDriven/mma_high_cpu_25nodes (53.72s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDataDriven.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/kv/kvserver/asim/state/new_state.go
+++ b/pkg/kv/kvserver/asim/state/new_state.go
@@ -384,19 +384,19 @@ func RangesInfoWithReplicaPlacement(
 	}
 
 	ret := initializeRangesInfoWithSpanConfigs(numRanges, config, minKey, maxKey, rangeSize)
-	rp.findReplicaPlacementForEveryStoreSet(numRanges)
+	result := rp.findReplicaPlacementForEveryStoreSet(numRanges)
 
 	rf := int(config.NumReplicas)
 
 	for rngIdx := 0; rngIdx < len(ret); rngIdx++ {
 		nextStoreSet := 0
-		for i := 0; i < len(rp); i++ {
-			if rp[i].Weight > 0 {
+		for i := 0; i < len(result); i++ {
+			if result[i].Ranges > 0 {
 				nextStoreSet = i
 				break
 			}
 		}
-		ratio := rp[nextStoreSet]
+		ratio := result[nextStoreSet]
 		if len(ratio.StoreIDs) != rf {
 			panic(fmt.Sprintf("expected %d replicas, got %d", rf, len(ratio.StoreIDs)))
 		}
@@ -410,7 +410,7 @@ func RangesInfoWithReplicaPlacement(
 			}
 		}
 		ret[rngIdx].Leaseholder = StoreID(ratio.LeaseholderID)
-		rp[nextStoreSet].Weight--
+		result[nextStoreSet].Ranges--
 	}
 	return ret
 }

--- a/pkg/kv/kvserver/asim/state/parser_replica_placement.go
+++ b/pkg/kv/kvserver/asim/state/parser_replica_placement.go
@@ -26,27 +26,37 @@ type ReplicaPlacement []Ratio
 // findReplicaPlacementForEveryStoreSet finds the replica placement for every
 // store set. We are overloading the Weight field here to calculate the actual
 // number of ranges to place with the given ratio.
-func (pr ReplicaPlacement) findReplicaPlacementForEveryStoreSet(numRanges int) {
+func (pr ReplicaPlacement) findReplicaPlacementForEveryStoreSet(
+	numRanges int,
+) []RatioWithRangesCount {
+	result := make([]RatioWithRangesCount, len(pr))
 	totalWeight := 0
 	for i := 0; i < len(pr); i++ {
 		totalWeight += pr[i].Weight
 	}
+	if totalWeight == 0 {
+		panic(fmt.Sprintf("zero total weight: %v", pr))
+	}
 	totalRangesToAllocate := numRanges
 	for i := 0; i < len(pr); i++ {
-		pr[i].Weight = int(float64(pr[i].Weight) * float64(numRanges) / float64(totalWeight))
-		totalRangesToAllocate -= pr[i].Weight
+		result[i] = RatioWithRangesCount{
+			Ratio:  pr[i],
+			Ranges: int(float64(pr[i].Weight) * float64(numRanges) / float64(totalWeight)),
+		}
+		totalRangesToAllocate -= result[i].Ranges
 	}
 	// Distribute the remaining ranges evenly across all ratios.
 	for i := 0; i < totalRangesToAllocate; i++ {
-		pr[i%len(pr)].Weight += 1
+		result[i%len(result)].Ranges += 1
 	}
+	return result
 }
 
-// Ratio struct to represent weight and store IDs
+// Ratio struct to represent weight and store IDs. All fields should be
+// read-only so that we can reuse the state across simulation runs.
 type Ratio struct {
 	// Weight is the relative weight of this ratio. It is used to calculate the
-	// number of ranges to place with the given ratio. findReplicaPlacementForEveryStoreSet
-	// later re-assigns the Weight field to the actual number of ranges to place.
+	// number of ranges to place with the given ratio.
 	Weight int
 	// StoreIDs is the list of store IDs to place replicas on.
 	StoreIDs []int
@@ -58,7 +68,21 @@ type Ratio struct {
 	LeaseholderID int
 }
 
-func (r Ratio) String() string {
+// RatioWithRangesCount wraps read-only Ratio with a mutable field to represent
+// the number of ranges to place with this ratio.
+// findReplicaPlacementForEveryStoreSet assigns the field based on the actual
+// number of ranges to place computed based on Ratio.
+type RatioWithRangesCount struct {
+	Ratio
+	// Ranges is the number of ranges to place with this ratio.
+	Ranges int
+}
+
+func (r RatioWithRangesCount) String() string {
+	return "{" + r.storeAndTypes() + "}:" + strconv.Itoa(r.Ranges)
+}
+
+func (r Ratio) storeAndTypes() string {
 	storeAndTypes := make([]string, len(r.StoreIDs))
 	for i := range r.StoreIDs {
 		s := "s" + strconv.Itoa(r.StoreIDs[i]) + ":"
@@ -73,7 +97,11 @@ func (r Ratio) String() string {
 		}
 		storeAndTypes[i] = s
 	}
-	return "{" + strings.Join(storeAndTypes, ",") + "}:" + strconv.Itoa(r.Weight)
+	return strings.Join(storeAndTypes, ",")
+}
+
+func (r Ratio) String() string {
+	return "{" + r.storeAndTypes() + "}:" + strconv.Itoa(r.Weight)
 }
 
 func (pr ReplicaPlacement) String() string {

--- a/pkg/kv/kvserver/asim/state/parser_replica_placement_test.go
+++ b/pkg/kv/kvserver/asim/state/parser_replica_placement_test.go
@@ -6,6 +6,7 @@
 package state
 
 import (
+	"strings"
 	"testing"
 
 	"github.com/cockroachdb/cockroach/pkg/testutils/datapathutils"
@@ -19,8 +20,16 @@ func TestReplicaPlacement(t *testing.T) {
 		datadriven.RunTest(t, path, func(t *testing.T, d *datadriven.TestData) string {
 			require.Equal(t, "parse", d.Cmd)
 			rp := ParseReplicaPlacement(d.Input)
-			rp.findReplicaPlacementForEveryStoreSet(1000)
-			return rp.String()
+			// Make sure rp is immutable by calling
+			// findReplicaPlacementForEveryStoreSet twice and comparing the results.
+			result1 := rp.findReplicaPlacementForEveryStoreSet(1000)
+			result2 := rp.findReplicaPlacementForEveryStoreSet(1000)
+			require.Equal(t, result1, result2)
+			var result []string
+			for _, r := range result1 {
+				result = append(result, r.String())
+			}
+			return strings.Join(result, "\n")
 		})
 	})
 }

--- a/pkg/kv/kvserver/asim/tests/testdata/non_rand/example_skewed_cpu_even_ranges_mma.txt
+++ b/pkg/kv/kvserver/asim/tests/testdata/non_rand/example_skewed_cpu_even_ranges_mma.txt
@@ -31,12 +31,29 @@ gen_load rate=100 rw_ratio=0 min_block=128 max_block=128 min_key=10001 max_key=2
 setting split_queue_enabled=false
 ----
 
-eval duration=25m samples=1 seed=42 cfgs=(mma-only) metrics=(cpu,leases,replicas,write_bytes_per_second)
+# TODO(tbg): it's interesting that sma-only does better on write throughput than
+# mma-only. Looking at the graphs, the mma-only flavor is much slower in moving
+# load around. Possibly a bug?
+eval duration=35m samples=1 seed=42 cfgs=(sma-only,mma-only,both) metrics=(cpu,leases,replicas,write_bytes_per_second)
 ----
-cpu#1: last:  [s1=569996683, s2=572459858, s3=577041089, s4=557385814, s5=555426485, s6=556145510, s7=570251533, s8=555498030, s9=557066732] (stddev=8256286.12, mean=563474637.11, sum=5071271734)
+cpu#1: last:  [s1=984642922, s2=1126669982, s3=703798726, s4=419425213, s5=432156047, s6=559114683, s7=142318086, s8=283346689, s9=424209084] (stddev=303225920.55, mean=563964603.56, sum=5075681432)
+leases#1: first: [s1=36, s2=0, s3=0, s4=36, s5=0, s6=0, s7=36, s8=0, s9=0] (stddev=16.97, mean=12.00, sum=108)
+leases#1: last:  [s1=13, s2=12, s3=12, s4=17, s5=11, s6=9, s7=15, s8=10, s9=9] (stddev=2.54, mean=12.00, sum=108)
+replicas#1: first: [s1=36, s2=36, s3=36, s4=36, s5=36, s6=36, s7=36, s8=36, s9=36] (stddev=0.00, mean=36.00, sum=324)
+replicas#1: last:  [s1=36, s2=37, s3=33, s4=37, s5=36, s6=36, s7=36, s8=35, s9=38] (stddev=1.33, mean=36.00, sum=324)
+write_bytes_per_second#1: last:  [s1=5619, s2=5639, s3=5275, s4=6269, s5=6137, s6=6150, s7=6187, s8=5861, s9=6290] (stddev=336.10, mean=5936.33, sum=53427)
+artifacts[sma-only]: 6f86408eff3edb14
+cpu#1: last:  [s1=572909386, s2=573979147, s3=572126524, s4=557253107, s5=557048324, s6=558469521, s7=571735501, s8=555522023, s9=554214626] (stddev=8138152.97, mean=563695351.00, sum=5073258159)
 leases#1: first: [s1=36, s2=0, s3=0, s4=36, s5=0, s6=0, s7=36, s8=0, s9=0] (stddev=16.97, mean=12.00, sum=108)
 leases#1: last:  [s1=4, s2=4, s3=4, s4=40, s5=4, s6=4, s7=40, s8=4, s9=4] (stddev=14.97, mean=12.00, sum=108)
 replicas#1: first: [s1=36, s2=36, s3=36, s4=36, s5=36, s6=36, s7=36, s8=36, s9=36] (stddev=0.00, mean=36.00, sum=324)
-replicas#1: last:  [s1=27, s2=28, s3=29, s4=40, s5=40, s6=40, s7=40, s8=40, s9=40] (stddev=5.68, mean=36.00, sum=324)
-write_bytes_per_second#1: last:  [s1=3757, s2=3916, s3=4016, s4=6906, s5=6916, s6=6921, s7=7006, s8=6996, s9=6983] (stddev=1443.42, mean=5935.22, sum=53417)
-artifacts[mma-only]: 14c792e14c656403
+replicas#1: last:  [s1=28, s2=28, s3=28, s4=40, s5=40, s6=40, s7=40, s8=40, s9=40] (stddev=5.66, mean=36.00, sum=324)
+write_bytes_per_second#1: last:  [s1=3898, s2=3891, s3=3878, s4=6939, s5=6939, s6=6929, s7=6990, s8=6964, s9=6978] (stddev=1446.16, mean=5934.00, sum=53406)
+artifacts[mma-only]: 1d73e264a60412ba
+cpu#1: last:  [s1=572415460, s2=566578888, s3=569096957, s4=556745916, s5=570328363, s6=558184663, s7=560070762, s8=557855218, s9=558141451] (stddev=5893908.39, mean=563268630.89, sum=5069417678)
+leases#1: first: [s1=36, s2=0, s3=0, s4=36, s5=0, s6=0, s7=36, s8=0, s9=0] (stddev=16.97, mean=12.00, sum=108)
+leases#1: last:  [s1=7, s2=10, s3=8, s4=17, s5=13, s6=13, s7=16, s8=12, s9=12] (stddev=3.13, mean=12.00, sum=108)
+replicas#1: first: [s1=36, s2=36, s3=36, s4=36, s5=36, s6=36, s7=36, s8=36, s9=36] (stddev=0.00, mean=36.00, sum=324)
+replicas#1: last:  [s1=36, s2=34, s3=36, s4=37, s5=36, s6=36, s7=37, s8=36, s9=36] (stddev=0.82, mean=36.00, sum=324)
+write_bytes_per_second#1: last:  [s1=5194, s2=5256, s3=5415, s4=6318, s5=6216, s6=6171, s7=6283, s8=6298, s9=6257] (stddev=461.69, mean=5934.22, sum=53408)
+artifacts[both]: 4fab04c2e6adb6a5

